backend/app/usecase/shortlink/updater_test.go

```diff
@@ -285,7 +293,7 @@ func TestShortLinkUpdaterPersist_UpdateShortLink(t *testing.T) {
 
 			shortLink, err := updater.UpdateShortLink(testCase.alias, testCase.update, testCase.user)
 			if testCase.expectedHasErr {
-				assert.NotEqual(t, nil, err)
+				assert.Equal(t, true, errors.As(err, &testCase.err))
```

Comment:
This checks they are both `Error` types but didn't verify what error it is.

Reply:
I'd argue that only the type of error matters for these test cases. It doesn't necessarily matter what the value inside the errors are, and checking the value in addition to the type could make tests more brittle.

Reply:
@byliuyang isn't that what we want. Ultimately we would like to have a specific error type for each error.

Reply:
@Coteh @arberiii I wasn't sure whether the error interface in the test case structure will prevent from verifying the real error type. I wonder whether we can experiment with it and see whether this work when we assign the concrete error  to the error interface.

Reply:
Sorry for not being clear.

Reply:
It is weird since I'm testing with testCase.err = ErrMaliciousLongLink and err  = ErrShortLinkNotFound, and the tests still passes

Reply:
I'm not sure why but something weird is happening with errors.As function. If I create two errors like this:
```go
err1 = ErrShortLinkNotFound
err2 = ErrMaliciousLongLink("HTTP")
errors.As(err1, &err2) // false
```
But if we have:
```go
err = ErrShortLinkNotFound
testCase.err = ErrMaliciousLongLink("HTTP")
errors.As(err2, &testCase.err) // true
```
Any idea what is happening?